bioptim/limits/multinode_constraint.py

```diff
@@ -0,0 +1,358 @@
+            ):
+                raise RuntimeError(
+                    "Impact transition without same nx is not possible, please provide a custom phase transition"
+                )
+
+            # Aliases
+            nlp_pre, nlp_post = all_pn[0].nlp, all_pn[1].nlp
+
+            # A new model is loaded here so we can use pre Qdot with post model, this is a hack and should be dealt
+            # a better way (e.g. create a supplementary variable in v that link the pre and post phase with a
+            # constraint. The transition would therefore apply to node_0 and node_1 (with an augmented ns)
+            model = biorbd.Model(nlp_post.model.path().absolutePath().to_string())
+
+            if nlp_post.model.nbContacts() == 0:
+                warn("The chosen model does not have any contact")
+            q_pre = nlp_pre.states["q"].mx
+            qdot_pre = nlp_pre.states["qdot"].mx
+            qdot_impact = model.ComputeConstraintImpulsesDirect(q_pre, qdot_pre).to_mx()
+
+            val = []
+            cx_end = []
+            cx = []
+            for key in nlp_pre.states:
+                cx_end = vertcat(cx_end, nlp_pre.states[key].mapping.to_second.map(nlp_pre.states[key].cx_end))
+                cx = vertcat(cx, nlp_post.states[key].mapping.to_second.map(nlp_post.states[key].cx))
+                post_mx = nlp_post.states[key].mx
+                continuity = nlp_post.states["qdot"].mapping.to_first.map(
+                    qdot_impact - post_mx if key == "qdot" else nlp_pre.states[key].mx - post_mx
+                )
+                val = vertcat(val, continuity)
+
+            name = f"PHASE_TRANSITION_{nlp_pre.phase_idx}_{nlp_post.phase_idx}"
+            func = biorbd.to_casadi_func(name, val, nlp_pre.states.mx, nlp_post.states.mx)(cx_end, cx)
+            return func
+
+        @staticmethod
+        def custom(multinode_constraint, all_pn, **extra_params):
+            """
+            Calls the custom transition function provided by the user
+
+            Parameters
+            ----------
+            multinode_constraint: MultinodeConstraint
+                A reference to the phase transition
+            all_pn: PenaltyNodeList
+                    The penalty node elements
+
+            Returns
+            -------
+            The expected difference between the last and first node provided by the user
+            """
+
+            nlp_pre, nlp_post = all_pn[0].nlp, all_pn[1].nlp
+            return multinode_constraint.custom_function(
+                multinode_constraint, nlp_pre, nlp_post, **extra_params
+            )
+
+
+class MultinodeConstraintFcn(Enum):
+    """
+    Selection of valid phase transition functions
+    """
+
+    CONTINUOUS = (MultinodeConstraintFunctions.Functions.continuous,)
+    IMPACT = (MultinodeConstraintFunctions.Functions.impact,)
+    CYCLIC = (MultinodeConstraintFunctions.Functions.cyclic,)
+    CUSTOM = (MultinodeConstraintFunctions.Functions.custom,)
```

Comment:
Remove those three

Then, PHASE_TRANSITION can inherits from `MultinodeConstraintFunctions` and implement them

Reply:
custom would have its place I guess.

Reply:
I'm not sure how to manage the inheritance.